pandas/core/frame.py

```diff
@@ -3745,7 +3745,7 @@ def _set_item_mgr(self, key, value: ArrayLike) -> None:
         # try to set first as we want an invalid
         # value exception to occur first
         if len(self):
-            self._check_setitem_copy()
+            self._check_setitem_copy(stacklevel=5)
```

Comment:
you can use the `find_stack_level` function i think its called

Reply:
yah at first i was trying to use find_stack_level sparingly, but might as well go for it

Reply:
Will do that as a follow up, since this would change the function signature I don't think we should backport this?

Reply:
sure that's fine